src/main/java/camera/CameraChestTracker.java

```diff
@@ -0,0 +1,87 @@
+package camera;
+
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfPoint;
+import org.opencv.core.Rect;
+import org.opencv.imgproc.Imgproc;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class for tracking chests.
+ */
+public class CameraChestTracker {
+
+    private Mat previousFrame = new Mat();
+    private  List<MatOfPoint> previousContours = new ArrayList<>();
+
+    /**
+     * Method which removes areas from frame, if they have overlap with the previous frame.
+     * @param frame a black and white (1s and 0s) single channel frame
+     * @param minChestArea The minimal area a rect has to be to be counted
+     * @return The frame with possible found areas removed
+     */
+    public Mat trackChests(final Mat frame, final double minChestArea) {
+        Mat tempFrame = frame.clone();
+
+        // calculate the contours in frame
+        List<MatOfPoint> contoursFrame = new ArrayList<>();
+        Mat contourMatFrame = new Mat();
+        Imgproc.findContours(tempFrame, contoursFrame,
+            contourMatFrame, Imgproc.RETR_CCOMP, Imgproc.CHAIN_APPROX_SIMPLE);
+
+        if (previousFrame.size().equals(frame.size())) {
+            // Check if there is overlap between chests in subsequent frames.
+            for (MatOfPoint points : contoursFrame) {
+                Rect rect = Imgproc.boundingRect(new MatOfPoint(points.toArray()));
+                if (rect.area() > minChestArea) {
+                    for (MatOfPoint points2 : previousContours) {
+                        Rect rect2 = Imgproc.boundingRect(new MatOfPoint(points2.toArray()));
+
+                        if (rect2.area() > minChestArea) {
+                            if (doOverlap(rect, rect2)) {
```

Comment:
An if in an if in a for in an if in a for in an if

Isn't there a way to make this better/cleaner?